### ISSUE TITLE
Detect when command arguments are not used and do not fail silently in such cases

### ISSUE THREAD

**Issue by ajsantander** [zeppelinos/zos#166]
Consider MyConract:
```
function initialize(address sender)  initializer public {...}
```

Right now, if you call a command like:
`npx zos create MyContract --init initialize 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1 --network development`

zOS will report the succesful creation of a proxy for MyContract. BUT, notice that the command above is incorrect, because it lacks the --args token. The correct way would be:
`npx zos create MyContract --init initialize --args 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1 --network development`

Depending on what's inside the initialize function, a user could spend hours trying to find what went wrong from within Solidity... If only zOS could notice that the argument 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1 was never used and fail with an error like "Unused command token 0x90f8bf6a479f320ead074411a4b0e7944ea8c9c1" or something.

Ideally, the solution should not only apply to the creation of proxies and arguments, but whenever a command syntax has "loose elements".

Another useful thing would be to trace out what initialize was called with when verbose is enabled.
In the case of the mistake above, it would have reported:
... initilize called with parameters: 0x0...

**Comment by spalladino** [comment]
@ajsantander does the contract have another `function initialize()` which receives no params, somewhere up in the inheritance chain?

**Comment by ajsantander** [comment]
Yes, MyContract is Ownable, whose initialize function receives no parameters.

**Comment by spalladino** [comment]
Got it! I was wondering how the tx had succeeded, but it seems that the CLI picked up the other initialize. The problem is specific to what you mention then: properly handling loose args.
